pkg/northbound/admin/admin.proto

```diff
@@ -142,6 +144,25 @@ message RollbackResponse {
     string message = 1;
 }
 
+// GetSnapshotRequest gets a snapshot
+message GetSnapshotRequest {
+    string device_id = 1 [(gogoproto.customname) = "DeviceID", (gogoproto.casttype) = "github.com/onosproject/onos-topo/pkg/northbound/device.ID"];
+}
+
+// ListSnapshotsRequest requests a list of snapshots
+message ListSnapshotsRequest {
+
+}
+
+// CompactChangesRequest requests a compaction of the change stores
+message CompactChangesRequest {
+    google.protobuf.Duration retention_period = 1 [(gogoproto.stdduration) = true];
+}
+
+// CompactChangesResponse is a compact response
+message CompactChangesResponse {
```

Comment:
maybe it could give a summary of how many network changes and device changes were compacted